ddtrace/internal/datadog/profiling/dd_wrapper/include/profiler_stats.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <cstddef>
+
+#include <string_view>
+
+namespace Datadog {
+
+class ProfilerState
+{
+  private:
+    static inline size_t sample_count = 0;
+
+  public:
+    ProfilerState() = default;
+    ~ProfilerState() = default;
+
+    static void increment_sample_count();
```

Comment:
I understand the aim was to have something stateless. 
NIT: I'd assume the API would have been increment(k_sample_count);